%description:
Tests that errors in finish() are properly caught and displayed in Cmdenv.

%module: Module

class Module : public cSimpleModule
{
  public:
    Module() : cSimpleModule(32768) { }
    virtual void activity() override;
    virtual void finish() override;
};

Define_Module(Module);

void Module::activity()
{
}

void Module::finish()
{
    throw std::runtime_error("something bad happened");
}

%exitcode: 1

%contains-regex: stderr
std::runtime_error: something bad happened


